src/io/memory.cr

```diff
@@ -418,23 +417,34 @@ class IO::Memory < IO
     io.write(to_slice)
   end
 
-  private def check_writeable
+  private def check_writeable : Nil
     unless @writeable
       raise IO::Error.new "Read-only stream"
     end
   end
 
-  private def check_resizeable
+  private def check_resizeable : Nil
     unless @resizeable
       raise IO::Error.new "Non-resizeable stream"
     end
   end
 
-  private def check_needs_resize
-    resize_to_capacity(@capacity * 2) if @bytesize == @capacity
+  private def check_needs_resize : Nil
+    resize_to_capacity(@capacity * GROWTH_FACTOR) if @bytesize == @capacity
+  end
+
+  private def resize_to_fit(size : Int32) : Nil
+    return if size <= capacity
+
+    capacity = @capacity * GROWTH_FACTOR
+    while size > capacity
+      capacity *= GROWTH_FACTOR
+    end
+
+    resize_to_capacity(capacity.to_i)
   end
 
-  private def resize_to_capacity(capacity)
+  private def resize_to_capacity(capacity : Int32) : Nil
```

Comment:
Is it safe to assume that capacity is an `Int32` (and not bigger)?

Reply:
I was wondering about that as well... I personally would set it to `UInt32` if not `UInt64`. Is there an equivalent to `size_t`?

Reply:
Currently, yes, it's limited to `Int32`. See #3209.